src/hotspot/share/runtime/stubDeclarations.hpp

```diff
@@ -0,0 +1,170 @@
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ *
+ */
+
+#ifndef SHARE_RUNTIME_SHAREDRUNTIME_ID_HPP
+#define SHARE_RUNTIME_SHAREDRUNTIME_ID_HPP
+
+#include "utilities/macros.hpp"
+
+// macros for generating definitions and declarations for shared, c1
+// and opto blob fields and associated stub ids
+
+// Different shared stubs can have different blob types and may
+// include some JFR stubs
+//
+// n.b resolve, handler and throw stubs must remain grouped in the
+// same order to allow id values to be range checked
+
+#if INCLUDE_JFR
+// template(name, type)
+#define SHARED_JFR_STUBS_DO(template)                                  \
+  template(jfr_write_checkpoint, RuntimeStub*)                         \
+  template(jfr_return_lease, RuntimeStub*)                             \
+
+#else
+#define SHARED_JFR_STUBS_DO(TEMPLATE)
+#endif
+
+// template(name, type)
+#define SHARED_STUBS_DO(template)                                      \
+  template(deopt, DeoptimizationBlob*)                                 \
+  /* resolve stubs */                                                  \
+  template(wrong_method, RuntimeStub*)                                 \
+  template(wrong_method_abstract, RuntimeStub*)                        \
+  template(ic_miss, RuntimeStub*)                                      \
+  template(resolve_opt_virtual_call, RuntimeStub*)                     \
+  template(resolve_virtual_call, RuntimeStub*)                         \
+  template(resolve_static_call, RuntimeStub*)                          \
+  /* handler stubs */                                                  \
+  template(polling_page_vectors_safepoint_handler, SafepointBlob*)     \
+  template(polling_page_safepoint_handler, SafepointBlob*)             \
+  template(polling_page_return_handler, SafepointBlob*)                \
+  /* throw stubs */                                                    \
+  template(throw_AbstractMethodError, RuntimeStub*)                    \
+  template(throw_IncompatibleClassChangeError, RuntimeStub*)           \
+  template(throw_NullPointerException_at_call, RuntimeStub*)           \
+  template(throw_StackOverflowError, RuntimeStub*)                     \
+  template(throw_delayed_StackOverflowError, RuntimeStub*)             \
+  /* other stubs */                                                    \
+  SHARED_JFR_STUBS_DO(template)                                        \
+
+// C1 stubs are always generated in a generic CodeBlob
+
+#ifdef COMPILER1
```

Comment:
c1 and Opto macros are not really needed in this PR so I'll remove them for now and re-add them in the follow-up PRs.